lib/utils.pm

```diff
@@ -1078,7 +1093,26 @@ sub set_hostname {
     assert_script_run "uname -n|grep $hostname";
     systemctl 'status network.service';
     save_screenshot;
-    assert_script_run "if systemctl -q is-active network.service; then systemctl reload-or-restart network.service; fi";
+
+    if (systemctl('is-active NetworkManager', ignore_failure => 1) == 0) {
+        my $conn_id = script_output 'basename $(ls -1 /etc/NetworkManager/system-connections | head -n 1) .nmconnection';
+
+        assert_script_run 'nmcli device down ' . $conn_id;
+        assert_script_run 'nmcli device up ' . $conn_id;
+
+        for (my $i = 0; $i < 30; $i++) {
+            my $state = script_output 'nmcli -w 30 networking connectivity check';
```

Comment:
i think the delay here is a bit overestimated. which would cause unnecessary slowness. Likely this will take 1 or 2 iterations.  if we have to iterate 30 times over it, would this make sense to set nmcli to 5 or 10?

Reply:
So far it has always taken one iteration. I guess if it has not connected after a few seconds then it is probably not going to connect.